packages/visual-editor/src/components/RootRenderer/DNDProvider.tsx

```diff
@@ -51,13 +56,17 @@ export const DNDProvider = ({ children }: Props) => {
     dragState.reset();
 
     if (!droppedItem.destination) {
-      if (!draggedItem?.destination) {
-        // User cancel drag
-        sendMessage(OUTGOING_EVENTS.ComponentDragCanceled);
-        return;
+      // User cancel drag
```

Comment:
I think its fairly confusing in here that we have a droppedItem and a draggedItem. Are those not the same thing?

Reply:
I refactored the code a bit and took the test code out and into its own component. Its still confusing to have these two seemingly like objects, though. I feel like this is fairly brittle and could break, especially if someone comes along not knowing to keep an eye out for both objects.

Reply:
Nice updates! I think renaming to `dropResult` helps.

When using click-to-drop in Firefox/Safari, the `draggedItem.destination` is always undefined (so it would cancel instead of dropping when the user clicks in the canvas).  Instead we can get the `dropResult.destination` to identify the dropzone destination when the user clicks the canvas to drop an item.

Now I'm wondering if we can rely on the `dropResult.destination` for drag-and-drop too 🤔. If `dropResult.destination` was reliable dropzone destination identifier for both click and drag, the code could be simplified a bit